src/components/spreadsheet/custom-columns/custom-columns-nodes-config.tsx

```diff
@@ -19,25 +25,130 @@ const styles = {
     },
 };
 
-type CustomColumnsNodesConfigProps = {
+enum NodesOptionId {
+    CONFIG = 'CONFIG',
+    REFRESH = 'REFRESH',
+}
+
+interface NodesOption {
+    id: NodesOptionId;
+    label: string;
+    action: () => void;
     disabled?: boolean;
-};
+    tooltipMsgId?: string;
+    tooltipMsgValues?: Record<string, any>;
+}
+
+interface CustomColumnsNodesConfigProps {
+    disabled?: boolean;
+    tabIndex: number;
+}
 
-export default function CustomColumnsNodesConfig({ disabled }: Readonly<CustomColumnsNodesConfigProps>) {
+export default function CustomColumnsNodesConfig({ disabled, tabIndex }: Readonly<CustomColumnsNodesConfigProps>) {
     const dialogOpen = useStateBoolean(false);
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const nodesAliases = useSelector((state: AppState) => state.customColumnsNodesAliases);
+    const currentNode = useSelector((state: AppState) => state.currentTreeNode);
+    const tableType = useSelector((state: AppState) => state.tables.definitions[tabIndex].type);
+    const studyUuid = useSelector((state: AppState) => state.studyUuid);
+    const currentRootNetworkUuid = useSelector((state: AppState) => state.currentRootNetworkUuid);
+
+    const { loadEquipmentData } = useLoadEquipment(
+        tableType,
+        studyUuid,
+        currentRootNetworkUuid,
+        currentNode?.id,
+        undefined
+    );
+
+    const nodesToReload = useMemo(() => {
+        // Get all aliased nodes ids, except for Root and current node (both are always up-to-date)
+        return nodesAliases.filter((node) => node.id !== currentNode?.id && node.name !== ROOT_NODE_LABEL);
+    }, [currentNode?.id, nodesAliases]);
+
+    const handleClick = useCallback((event: MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    }, []);
+
+    const handleClose = useCallback(() => {
+        setAnchorEl(null);
+    }, []);
+
+    const handleRefresh = useCallback(() => {
+        if (nodesToReload?.length) {
+            const promises = loadEquipmentData(nodesToReload.map((n) => n.id as string));
+            Promise.all(promises).then(() => {
+                console.debug(`Refresh done for ${promises.length} nodes among ${nodesToReload.length} aliased nodes`);
+            });
+        }
+    }, [loadEquipmentData, nodesToReload]);
```

Comment:
Same goes for this

Reply:
the reason I return the promises to the caller : he can perform a specific action when all promises end. In the case above, it's only a console log (we could remove it). In the case of the useLoadEquipment we reset a state (cf _setIsFetching(false)_).

Maybe I could add a callback _onFetchingDone_ to allow specific action from the caller ?

Reply:
Yes could be nice :)